bindings/java/mongocrypt/build.gradle.kts

```diff
@@ -0,0 +1,279 @@
+                | =================
+                |
+                | $version != $gitVersion
+                |
+                | The project version does not match the git tag.
+                |""".trimMargin()
+            throw GradleException(cause)
+        } else {
+            println("Publishing: ${project.name} : $gitVersion")
+        }
+    }
+
+    if (gitVersion == version) {
+        dependsOn(tasks.withType<PublishToMavenRepository>())
+    }
+}
+
+/*
+ * jnaLibs check - before all publishing tasks
+ */
+val jnaLibsPath = System.getProperty("jnaLibsPath", "")
+tasks.withType<AbstractPublishToMaven> {
+    if (!System.getProperties().containsKey("jnaLibsNoCheck")) {
+        doFirst {
+            if (jnaLibsPath.isEmpty()) {
+                val cause = """
+                |
+                | No Jna Libs:
+                | ============
+                |
+                | Missing Jna Libs - pubishing would not include any libs!
+                |
+                | Please include -DjnaLibsPath=<path-to-libs> when running a publish command.
+                |""".trimMargin()
+                throw GradleException(cause)
+            } else {
+                println("\n\nPlease confirm the jnaLibs resources layout: (Use -DjnaLibsNoCheck to bypass)\n")
+                File(jnaLibsPath).walkTopDown().forEach { println(" - ${it.toString().removePrefix(jnaLibsPath)}") }
+                println("--------------------------------------------")
+                println("Is the above the expected layout? [y/n?]\n\n")
+                if (readLine()!!.trim().toLowerCase() != "y") {
+                    throw GradleException("Cancelling...")
+                }
+            }
+        }
+    }
+
+    if (jnaLibsPath.isNotEmpty()) {
+        sourceSets["main"].resources.srcDirs("resources", jnaLibsPath)
+    }
+}
+
+
+/*
+For security we allow the signing-related project properties to be passed in as environment variables, which
+Gradle enables if they are prefixed with "ORG_GRADLE_PROJECT_".  But since environment variables can not contain
+the '.' character and the signing-related properties contain '.', here we map signing-related project properties with '_'
+to ones with '.' that are expected by the signing plugin.
+*/
+gradle.taskGraph.whenReady {
+    if (allTasks.any { it is Sign }) {
+        val signing_keyId: String? by project
+        val signing_secretKeyRingFile: String? by project
+        val signing_password: String? by project
+
+        allprojects {
+            signing_keyId?.let { extra["signing.keyId"] = it }
+            signing_secretKeyRingFile?.let { extra["signing.secretKeyRingFile"] = it }
+            signing_password?.let { extra["signing.password"] = it }
+        }
+    }
+}
+
+tasks.withType<Test> {
+    @Suppress("UNCHECKED_CAST")
+    systemProperties((System.getProperties().toMap() as Map<String, Any>).filter { it.key.startsWith("jna.") })
+}
+
+/*
+ * Jna Libs - temporary testing helpers
```

Comment:
Seems like the right thing to do for testing is just to build locally from the cmake-build directory, rather than downloading.  This is what we do now in Evergreen and seems like the right thing, as you're guaranteed to be in sync and you only need the platform that you're actually running the tests on. 

 I think we only need to download files when we are publishing a snapshot or a release.